miro_preprocessor/miro_inventory/src/miro_inventory.py

```diff
@@ -75,4 +76,17 @@ def main(event, _):
         json.dumps(indexable)
     )
     print(f'_put_to_es response: {response}')
-    response.raise_for_status()
+
+    # A very common source of errors is a 429: Rate Limited Exceeded from
+    # the Elasticsearch cluster.  If we raise an exception here, the Lambda
+    # will be rescheduled almost immediately and go back around to put more
+    # load on the cluster.  So if we detect an error, insert an artificial
+    # delay to slow down the Lambda rescheduler, and reduce cluster pressure.
+    #
+    # Note: if this hits the timeout instead, the Lambda will still register
+    # as failed and be rescheduled, so this doesn't cause us to lose events.
+    try:
+        response.raise_for_status()
+    except Exception:
+        time.sleep(10)
+        raise
```

Comment:
This Lambda currently has a timeout of 120s (!), so I don’t think this will make much of a difference.

Reply:
I don't think this is necessary since the cluster size has been increased. The solution is to _run less through the pipeline in any case. Also what @alexwlchan said.